hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java

```diff
@@ -31,99 +31,189 @@
+            "Tag value cannot be null.",
+            position = 10)
     @JsonInclude(Include.NON_EMPTY)
     protected Map<String, String> tags;
 
     /** A list of links to actions represented by TriggerAction*/
+    @ApiModelProperty(value = "A list of links to actions.",
+            position = 11)
     @JsonInclude(Include.NON_EMPTY)
     private Set<TriggerAction> actions;
 
     /** Disable automatically after firing */
+    @ApiModelProperty(value = "Disable automatically after firing.",
+            position = 12,
+            example = "false")
     @JsonInclude
     private boolean autoDisable;
 
     /** Enable automatically if disabled and resolved manually */
+    @ApiModelProperty(value = "Enable automatically if disabled and resolved manually.",
+            position = 13,
+            example = "false")
     @JsonInclude
     private boolean autoEnable;
 
     /** Switch to auto-resolve mode after firing */
+    @ApiModelProperty(value = "Switch to auto-resolve mode after firing.",
+            position = 14,
+            example = "false")
     @JsonInclude
     private boolean autoResolve;
 
     /** Resolve all unresolved alerts when auto-resolve condition-set is satisfied */
+    @ApiModelProperty(value = "Resolve all unresolved alerts when auto-resolve condition-set is satisfied.",
+            position = 15,
+            example = "false")
     @JsonInclude
     private boolean autoResolveAlerts;
 
+    @ApiModelProperty(value = "The policy used for deciding whether the trigger auto-resolved condition-set is " +
+            "satisfied. Either ALL conditions must evaluate to true, or whether ANY one is enough.",
+            position = 16,
+            example = "ALL")
     @JsonInclude
     private Match autoResolveMatch;
 
     // Only set for MEMBER triggers, the dataIdMap used when adding the member. Is re-used
     // for group condition updates unless a new dataIdMap is provided.
+    @ApiModelProperty(value = "Only set for MEMBER triggers, the dataIdMap used when adding the member. " +
+            "It is reused for group condition updates unless a new dataIdMap is provided.",
+            position = 17)
     @JsonInclude(Include.NON_EMPTY)
     protected Map<String, String> dataIdMap;
 
     // Only set for MEMBER triggers, the group trigger for which this is a member.
+    @ApiModelProperty(value = "Only set for MEMBER triggers, the group trigger for which this is a member.",
+            position = 18)
     @JsonInclude(Include.NON_EMPTY)
     private String memberOf;
 
+    @ApiModelProperty(value = "A enabled trigger is loaded into the engine for data evaluation.",
+            position = 19,
+            example = "false")
     @JsonInclude
     private boolean enabled;
 
+    @ApiModelProperty(value = "The policy used for deciding whether the trigger condition-set is satisfied. " +
+            "Either ALL conditions must evaluate to true, or whether ANY one is enough.",
```

Comment:
Suggestion:
'Either ALL conditions must evaluate to true, or whether ANY one is enough.'
->
'ALL conditions must evaluate to true or ANY one condition must evaluate to true.'